src/renderer/features/dynamicDerivations/ImportKeysModal/ImportKeysModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { useUnit } from 'effector-react';
+import { useEffect } from 'react';
+
+import { Alert, BaseModal, Button, InfoLink, InputFile, InputHint } from '@renderer/shared/ui';
+import { useI18n } from '@renderer/app/providers';
+import { importKeysModel, TypedImportedDerivation } from '@renderer/entities/dynamicDerivations';
+import { AccountId } from '@renderer/shared/core';
+import { cnTw } from '@renderer/shared/lib/utils';
+// @ts-ignore
+import templateFile from '@renderer/assets/files/dd-template.yaml';
+
+type Props = {
+  isOpen: boolean;
+  rootAccountId: AccountId;
+  existingKeys: TypedImportedDerivation[];
+  onClose: () => void;
+};
+
+export const ImportKeysModal = ({ isOpen, onClose, rootAccountId, existingKeys }: Props) => {
+  const { t } = useI18n();
+  const validationError = useUnit(importKeysModel.$validationError);
+  const successReport = useUnit(importKeysModel.$successReport);
+
+  useEffect(() => {
+    if (rootAccountId && existingKeys) {
```

Comment:
`Props` have these parameters as required, why do we check them?
`ImportKeysModal` interfaces states what we must provide, it's client's job to check that everything is in right place

Reply:
removed check here